crates/types/src/simple_certificate.rs

```diff
@@ -145,18 +147,103 @@ impl<TYPES: NodeType, VOTEABLE: Voteable + 'static, THRESHOLD: Threshold<TYPES>>
             _pd: PhantomData,
         }
     }
-    async fn is_valid_cert<MEMBERSHIP: Membership<TYPES>, V: Versions>(
+    async fn is_valid_cert<V: Versions>(
         &self,
+        stake_table: Vec<<TYPES::SignatureKey as SignatureKey>::StakeTableEntry>,
```

Comment:
not sure if this is immediately fixable (might require epoch numbers in the certs, which is being worked on separately), but I think `is_valid_cert` should actually just take the `Memberships` directly, rather than the stake table + threshold

we always know which stake table/threshold to use for each cert

can you make an issue for this and add it to the board

Reply:
Before this PR that was the logic. We changed it b/c `is_valid_cert` can't be generic of `Membership` b/c we have to call specific functions such `stake_table` or `da_stake_table`. This PR addresses this by passing in the stake table.

Reply:
https://github.com/EspressoSystems/HotShot/issues/3918#issue-2690133601